src/main/java/com/google/sps/data/User.java

```diff
@@ -1,57 +1,195 @@
-  private String email;
-  private String id;
-  private String username;
-  // shopping list will not be implented until after MVP
-  private ArrayList<String> shoppingList;
+  // API variables:
+  // client ID is generated from Ali's google APIs credentials page.
+  private static String CLIENT_ID = "1034390229233-u07o0iaas2oql8l4jhe7fevpfsbrtv7n.apps.googleusercontent.com";
+  // Payload class contains decrypted user information.
+  private Payload payload;
+  private Entity entity;
 
-  public User(String id, String email, String username) {
-    this.id = id;
-    this.email = email;
-    this.username = username;
-    planner = new ArrayList<Long>();
-    cookbook = new ArrayList<Long>();
-    userRecipes = new ArrayList<Long>();
+  /**
+    // User Entity Properties:
+
+    "displayName" : "Example Name"
+    "cookbook" : [ "longId1", "longId2", "longId3" ]
+    "userRecipes" : [ "longId1", "longId2", "longId3" ]
+    "planner" : [ "longId1", "longId2", "longId3" ]
+    "shoppingList" : [ "longId1", "longId2", "longId3" ]
+
+    // Properties that can be accessed from Payload object:
+
+    // These seven fields are only included when the user has granted the "profile" and
+    // "email" OAuth scopes to the application.
+    // "sub" or Subject refers to the user-unique ID number.
+    "sub" : "110169484474386276334"
+    "email": "testuser@gmail.com",
+    "email_verified": "true",
+    "name" : "Test User",
+    "picture": "https://lh4.googleusercontent.com/-kYgzyAWpZzJ/ABCDEFGHI/AAAJKLMNOP/tIXL9Ir44LE/s99-c/photo.jpg",
+    "given_name": "Test",
+    "family_name": "User",
+    "locale": "en"
+  */
+
+  /**
+   * Prevents usage of default constructor
+   */
+  private User() { }
+
+  /**
+   * Constructs a User instance based on a unique id_token.
+   * Verify user using google's ID token verifyer API.
+   * This method is commented out because it does not work yet.
+   */
+  public User(String idTokenString) {
+    // GoogleIdTokenVerifier verifier = new GoogleIdTokenVerifier
+    // .Builder(UrlFetchTransport.getDefaultInstance(), new JsonFactory())
+    // // Specify the CLIENT_ID of the app that accesses the backend:
+    // .setAudience(Collections.singletonList(CLIENT_ID)).build();
+
+    // GoogleIdToken idToken = verifier.verify(idTokenString);
+    // if (idToken == null) {
+    //   throw new Exception("ID token verification failed");
+    // }
+
+    // payload = idToken.getPayload();
+  }
+
+  /**
+   * This creates a standard universal user instance regardless of login status.
+   * This test method will not be removed from the webapp prior to deployment.
+   */
+  public static User getTestUser() {
+    User testUser = new User();
+    String testID = "000000000000000000000";
+    Key userKey = KeyFactory.createKey("User", testID);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    try {
+        testUser.entity = datastore.get(userKey);
+    } catch(Exception e) {
```

Comment:
The style guide isn't clear here and I'm not totally sure, but if I remember correctly, I believe it's best practice to not use a general exception in Java, but rather to catch specific errors and handle them . It looks like you already know the name of the exception, so including it could be a small change.